lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2510,7 +2510,7 @@ LogicalResult ConvertAtenOp<AtenUnsqueezeOp>::matchAndRewrite(
   if (!matchPattern(op.getDim(), m_TorchConstantInt(&dim)))
     return rewriter.notifyMatchFailure(op, "dim must be a Scalar constant");
 
-  dim = toPositiveDim(dim, selfRank);
+  dim = toPositiveDim(dim, selfRank + 1);
```

Comment:
I recommend adding a comment stating how unsqueeze cannot use the native toPositiveDim semantics and has special needs.

Reply:
Done.